eval/src/tests/instruction/join_with_number/join_with_number_function_test.cpp

```diff
@@ -51,6 +51,8 @@ void verify_optimized(const vespalib::string &expr, Primary primary, bool inplac
     const auto stable_types = CellTypeSpace({CellType::FLOAT, CellType::DOUBLE}, 2);
     FunInfo stable_details{primary, inplace};
     TEST_DO(EvalFixture::verify<FunInfo>(expr, {stable_details}, stable_types));
+    const auto unstable_types = CellTypeSpace({CellType::BFLOAT16, CellType::INT8}, 2);
+    TEST_DO(EvalFixture::verify<FunInfo>(expr, {}, unstable_types));
```

Comment:
We probably want this optimizer to also apply to unstable cells, since it is one of the usual ways to handle quantized values (multiply with a constant). I guess the issue is with template instantiation due to sharing code between inplace and non-inplace cases...